lib/spack/spack/build_systems/sip.py

```diff
@@ -157,7 +158,8 @@ def configure(self, pkg, spec, prefix):
             ]
         )
 
-        self.python(configure, *args)
+        python = Executable(spec["python"].prefix.bin.python)
+        python(configure, *args)
```

Comment:
I think this should work:
```suggestion
        self.pkg.python(configure, *args)
```